internal/fs/fs.go

```diff
@@ -980,7 +980,9 @@ func (fs *fileSystem) lookUpOrCreateChildInode(
 	parent inode.DirInode,
 	childName string) (child inode.Inode, err error) {
 	// First check if the requested child is a localFileInode.
-	child = fs.lookUpLocalFileInode(parent, childName)
+	if child, err = fs.lookUpLocalFileInode(parent, childName); err != nil {
+		return child, err
+	}
 	if child != nil {
```

Comment:
should we just combine both the conditions into one line.
if child != nil | err != nil {
return child, err
}

Reply:
I'd find that hard to read - since it captures both a success scenario as well as a failure scenario.